File: Theory + Practice/Olymp/Oblast 2016/Day2a/Day2/sources_all/T2Z1NA02.pas
//Kamenko Pavel
var ost,i,t,s,b,m,w,ans:longint;
    d:array[1..100010]of longint;
    f:text;
begin
 assign(f,'input.txt');
 reset(f);
 read(f,m,w);
 for i:=1 to m do
  read(f,d[i]);
 read(f,s,b,t);
 close(f);
 ost:=s;
 for i:=1 to m do
  begin
   if d[i]<t
    then begin
          ost:=(ost+d[i])mod w;
          if ost=0
           then ost:=w;
         end
    else begin
          if (ost+t-1) mod w =0
           then begin if b=w
                 then inc(ans)
                end
           else
            if (ost+t-1) mod w= b
             then inc(ans);
          ost:=(ost+d[i]) mod w;
          if ost=0
           then ost:=w;
         end;
  end;
assign(f,'output.txt');
rewrite(f);
write(f,ans);
close(f);



end.

File: Theory + Practice/Olymp/Oblast 2015/Day2/Day2/Solutions/T2Z1NA02.cpp
#include <fstream>
#include <string>
using namespace std;
string s1,s2;
int l1,l2,ans;
int main() {
	ifstream f1("input.txt");
	ofstream f2("output.txt");
	f1 >> s1 >> s2;
	l1 = s1.length() - 1; l2 = s2.length() - 1;
	ans = 0;
	while(l1 != -1 && l2 != -1){
		if(s1[l1] == s2[l2]) ans++; else break;
		l1--; l2--;
	}
	f2 << ans;
	f1.close(); f2.close();
	return 0;
}
